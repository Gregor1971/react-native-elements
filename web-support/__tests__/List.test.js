import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../samples';
import { List as Component, ListItem } from '../../src';

const children = [
  <ListItem key="v0" title="item 1" />,
  <ListItem key="v1" title="item 2" />,
  <ListItem key="v2" title="item 3" />,
];

const props = {
  'no props': {
    component: Component,
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  containerStyle: {
    component: Component,
    children,
    props: {
      containerStyle: { borderWidth: 5 },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Lists: { sectionComponents: { List: { samples } } },
  },
  options
);
