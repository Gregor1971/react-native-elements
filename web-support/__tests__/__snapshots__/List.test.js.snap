// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists: List: [ prop: containerStyle ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      77,
      Object {
        "borderWidth": 5,
      },
    ]
  }
>
  <ListItem
    chevronColor="#bdc6cf"
    disabled={false}
    hideChevron={false}
    leftIconUnderlayColor="white"
    rightIcon={
      Object {
        "name": "chevron-right",
      }
    }
    rightTitleNumberOfLines={1}
    roundAvatar={false}
    subtitleNumberOfLines={1}
    switchButton={false}
    textInputEditable={true}
    title="item 1"
    titleNumberOfLines={1}
    underlayColor="white"
  />
  <ListItem
    chevronColor="#bdc6cf"
    disabled={false}
    hideChevron={false}
    leftIconUnderlayColor="white"
    rightIcon={
      Object {
        "name": "chevron-right",
      }
    }
    rightTitleNumberOfLines={1}
    roundAvatar={false}
    subtitleNumberOfLines={1}
    switchButton={false}
    textInputEditable={true}
    title="item 2"
    titleNumberOfLines={1}
    underlayColor="white"
  />
  <ListItem
    chevronColor="#bdc6cf"
    disabled={false}
    hideChevron={false}
    leftIconUnderlayColor="white"
    rightIcon={
      Object {
        "name": "chevron-right",
      }
    }
    rightTitleNumberOfLines={1}
    roundAvatar={false}
    subtitleNumberOfLines={1}
    switchButton={false}
    textInputEditable={true}
    title="item 3"
    titleNumberOfLines={1}
    underlayColor="white"
  />
</View>
`;

exports[`Lists: List: [ prop: no props ] enzyme-mount: snapshot 1`] = `
<List>
  <View
    style={
      Array [
        77,
        undefined,
      ]
    }
  >
    <div
      className="rn-alignItems-1oszu61 rn-backgroundColor-44z8sh rn-borderTopColor-gv7t1l rn-borderRightColor-1itbhdy rn-borderBottomColor-1u5ghgs rn-borderLeftColor-592z09 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-5kkj8d rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-marginTop-1x0uki6 rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
    />
  </View>
</List>
`;

exports[`Lists: List: [ prop: no props ] enzyme-render: snapshot 1`] = `
<div
  class="rn-alignItems-1oszu61 rn-backgroundColor-44z8sh rn-borderTopColor-gv7t1l rn-borderRightColor-1itbhdy rn-borderBottomColor-1u5ghgs rn-borderLeftColor-592z09 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-5kkj8d rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-marginTop-1x0uki6 rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
/>
`;

exports[`Lists: List: [ prop: no props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Array [
      77,
      undefined,
    ]
  }
/>
`;
